server/auth/login-gov.es6

```diff
@@ -16,73 +17,60 @@ let basicAuthOptions = {
   }
 };
 
-loginGov.setup = () => {
-  console.log('------------ in loginGov.setup ', vcapServices.loginGovIdpUsername, vcapServices.loginGovIdpPassword);
-  Issuer.defaultHttpOptions = basicAuthOptions;
-  Issuer.discover('https://idp.int.login.gov/.well-known/openid-configuration')
-    .then(loginGovIssuer => {
-      console.log('----- loginGovIssuer: ', loginGovIssuer);
+const params = {
+  acr_values: 'http://idmanagement.gov/ns/assurance/loa/1',
+  nonce: `${Math.random()}-${Math.random()}`,
+  prompt: 'select_account',
+  // TODO: replace with VCAP_SERVICES
+  redirect_uri: 'https://fs-intake-api-login-test.app.cloud.gov/auth/login-gov/openid/callback',
+  response_type: 'code',
+  scope: 'openid email',
+  state: `${Math.random()}-${Math.random()}`
+};
 
-      let client = new loginGovIssuer.Client({
-        client_id: vcapServices.loginGovIssuer,
-        client_secret: 'thisisaveryspecialsecret'
-      });
+passport.serializeUser((user, done) => {
+  done(null, user.email);
+});
 
-      let params = {
-        redirect_uri: 'https://fs-intake-api-staging.app.cloud.gov/auth/login-gov/openid/callback',
-        acr_values: 'http://idmanagement.gov/ns/assurance/loa/1',
-        response_type: 'code',
-        scope: 'openid email',
-        prompt: 'select_account',
-        nonce: '12345678901234567890123456789012'
-      };
+passport.deserializeUser((email, done) => {
+  done(null, { email: email });
+});
 
+loginGov.setup = () => {
+  Issuer.defaultHttpOptions = basicAuthOptions;
+  Issuer.discover('https://idp.int.login.gov/.well-known/openid-configuration').then(loginGovIssuer => {
+    // don't use the userinfo_endpoint, as the userinfo payload is returned with the token_id
+    delete loginGovIssuer.userinfo_endpoint;
+    let keys = { keys: [vcapServices.loginGovJwk] };
+    jose.JWK.asKeyStore(keys).then(joseKeystore => {
+      let client = new loginGovIssuer.Client(
```

Comment:
Wow, this is not intuitive at all. No wonder we had such a hard time figuring it out. As such, I would like a lot more documentation on this.